### ISSUE TITLE
[BUG] Is flash attention 3 compatible with Ulysses or not?

### ISSUE THREAD

**Issue by kisseternity** [deepspeedai/DeepSpeed#7513]
**Describe the bug**
Hello, I've been training Qwen3 recently with FA3 and Ulysses sp > 1, but I get nan logits after forward:
(WorkerDict pid=8874) WARN: rank 1 non-finite policy loss. pg_loss_finite=False ppo_kl_finite=False. old_log_prob_stats={'shape': (3, 4096), 'nan': 136, 'inf': 0, 'min': -26.433170318603516, 'max': 0.0} log_prob_stats={'shape': (3, 4096), 'nan': 0, 'inf': 0, 'min':

**To Reproduce**
1. Using verl for dapo training.
2. Using fa3 and Ulysses sp > 1.
3. Observing that loss is nan.
4. Print forward logits and find out that those are already nan.

**Expected behavior**
Loss should be not nan.

**Screenshots**

<img width="2697" height="647" alt="Image" src="https://github.com/user-attachments/assets/27f1183e-8015-4737-b4b0-b3dee7d6daf0" />

**System info (please complete the following information):**
 - OS: [e.g. Ubuntu 18.04]
 - GPU count and types [e.g. two machines with x8 A100s each]
 - Interconnects (if applicable) [e.g., two machines connected with 100 Gbps IB]
 - Python version
 - Any other relevant info about your setup
